x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/support/ReloadableSecurityComponent.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.security.support;
+
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.plugins.ReloadablePlugin;
+
+/**
+ * This interface allows adding support for reload operations (on secure settings change) in a generic way fore security components.
+ * The implementors of this interface will be called when the values of {@code SecureSetting}s should be reloaded by security plugin.
+ * For more information about reloading plugin secure settings, see {@link ReloadablePlugin}.
+ */
+public interface ReloadableSecurityComponent {
```

Comment:
This change was inspired by Tim's [POC](https://github.com/tvernum/elasticsearch/commit/reload-ldap-password). I liked the generic approach and decided to adopt it.